h2o-r/h2o-package/R/automl.R

```diff
@@ -580,3 +579,84 @@ h2o.get_leaderboard <- function(object, extra_columns=NULL) {
   if (!.is.H2OAutoML(object)) stop("Only H2OAutoML instances are currently supported.")
   return(.automl.fetch_leaderboard(attr(object, 'id'), extra_columns))
 }
+
+
+#' Get best model of a given family/algorithm from an AutoML object.
+#'
+#' @param object H2OAutoML object
+#' @param algorithm One of "base_model", "deep_learning", "drf", "gbm", "glm", "stacked_ensemble", "xgboost", "xrt"
+#' @param criterion Criterium can be one of the metrics reported in leaderboard, if NULL pick the first metric
+#'                  for each task from the following list:
+#'                  * Regression metrics: mean_residual_deviance, rmse, mse, mae, rmsle
+#'                  * Binomial metrics: auc, logloss, aucpr, mean_per_class_error, rmse, mse
+#'                  * Multinomial metrics: mean_per_class_error, logloss, rmse, mse, auc, aucpr
+#' @return a model or None if none of a given family is present
+#' @examples
+#' \dontrun{
+#' library(h2o)
+#' h2o.init()
+#' prostate_path <- system.file("extdata", "prostate.csv", package = "h2o")
+#' prostate <- h2o.importFile(path = prostate_path, header = TRUE)
+#' y <- "CAPSULE"
+#' prostate[,y] <- as.factor(prostate[,y])  #convert to factor for classification
+#' aml <- h2o.automl(y = y, training_frame = prostate, max_runtime_secs = 30)
+#' gbm <- h2o.get_best_model(aml, "gbm")
+#' }
+#' @export
+h2o.get_best_model <- function(object,
+                               algorithm = c("base_model", "deep_learning", "drf", "gbm",
+                                             "glm", "stacked_ensemble", "xgboost", "xrt"),
+                               criterion = NULL) {
+  .get_best <- function (leaderboard, pattern, criterion, ascending) {
+    leaderboard <- do.call(h2o.arrange, list(
+      leaderboard,
+      if(ascending) criterion else bquote(desc(.(criterion)))))
```

Comment:
I am not happy with this: `h2o.arrange(frame, ...)`  delays the evaluation of the `...` to mimic the functions like `dplyr::arrange` but doing it in base R way (not `rlang`/`tidyeval` way as `dplyr`) which makes it hard to use in non-interactive way, e.g., using a variable to pick the column name to sort by.

One reason for quoting/delaying the evaluation is the `desc` which isn't defined.

AFAIK using functions that behave like this (e.g. `base::subset`) in non-interactive code is usually frowned upon so if there is another way I think it would be preferable but I didn't find any yet.